scripts/release/workflow/publish.sh

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env bash
+
+set -o errexit
+
+package_name() {
+  echo $(node --print --eval "require('./package.json').name")
+}
+
+latest_npm_version() { 
+  echo $(npm info "$package_name" version)
+}
+
+package_json_version() {
+  echo $(node --print --eval "require('./package.json').name")
+}
+
+dist_tag() {
+  if [ $PRERELEASE = "true" ]; then
+    echo "next"
+  else
+    if [ `npx semver -r ">$package_json_version" $latest_npm_version` = "" ]; then
+      echo "latest"
+    else
+      # This is a patch for an older version
+      # npm can't publish without a tag
+      echo "tmp"
+    fi
+  fi
+}
+
+publish() {
+  cd contracts
+
+  ## Intentionally escape $ to avoid interpolation and to write the token to disk
+  echo "//registry.npmjs.org/:_authToken=\${NPM_TOKEN}" > .npmrc
```

Comment:
FYI you can also avoid interpolation by using single quotes. This is fine though. Especially the comment, it's good to make it explicit.